Add S3 Gateway Endpoints to the Private Networks for the OpenVPN Server

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR adds the prefix list for the S3 gateway endpoint created in https://github.com/cisagov/cool-sharedservices-networking/pull/47.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

These routes are needed to ensure that AWS S3 traffic is sent over the VPN. This is specifically needed so VPN users can access buckets with policies such as what was created in https://github.com/cisagov/cool-images-assessment-images/pull/4. @jsf9k suggested we try and pull them from the remote state for [cisagov/cool-sharedservices-networking](https://github.com/cisagov/cool-sharedservices-networking) so that these routes are not manually maintained using `tfvars`.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
@jsf9k confirmed that there we no changes by testing an apply against the manually added routes he deployed in the Staging environment to test bucket access.
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

mcdonnnj committed:
Add S3 gateway endpoint CIDRs to the OpenVPN routes

Take the S3 gateway endpoint CIDRs, convert them to "network netmask" entries,
and concatenate that list with the `private_networks` variable to pass as the
`private_networks` argument of the `openvpn` module. This will allow traffic
that is destined for the S3 service in the deployed region to be routed over
the VPN connection.

mcdonnnj committed:
Update the `private_networks` variable description

Update the `private_networks` variable description to mention that it is
concatenated with the S3 gateway endpoint CIDRs.

jsf9k commented on README.md:
```diff
@@ -40,7 +40,7 @@ has been applied.
 | Name | Source | Version |
 |------|--------|---------|
-| openvpn | github.com/cisagov/openvpn-server-tf-module?ref=improvement%2Flink-nessus-agent |  |
+| openvpn | github.com/cisagov/openvpn-server-tf-module |  |
```
Thanks for catching this and cleaning it up!

jsf9k commented on README.md:
```diff
@@ -75,7 +75,7 @@ has been applied.
 | nessus\_key\_key | The SSM Parameter Store key whose corresponding value contains the secret key that the Nessus Agent should use when linking with the CDM Tenable Nessus server (e.g. /cdm/nessus/key). | `string` | `"/cdm/nessus_key"` | no |
 | nessus\_port\_key | The SSM Parameter Store key whose corresponding value contains the port to which the Nessus Agent should connect when linking with the CDM Tenable Nessus server (e.g. /cdm/nessus/port). | `string` | `"/cdm/nessus_port"` | no |
-| private\_networks | A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. ["10.224.0.0 255.240.0.0", "192.168.100.0 255.255.255.0"]).  These will be pushed to the clients. | `list(string)` | n/a | yes |
+| private\_networks | A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. ["10.224.0.0 255.240.0.0", "192.168.100.0 255.255.255.0"]).  This will be concatenated with the list of S3 gateway endpoint routes and these will be pushed to the clients. | `list(string)` | n/a | yes |
```
```suggestion
| private\_networks | A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. ["10.224.0.0 255.240.0.0", "192.168.100.0 255.255.255.0"]).  This will be concatenated with the list of S3 gateway endpoint routes and the result will be pushed to the clients. | `list(string)` | n/a | yes |
```

jsf9k commented on variables.tf:
```diff
@@ -68,7 +68,7 @@ variable "nessus_port_key" {
 variable "private_networks" {
   type        = list(string)
-  description = "A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. [\"10.224.0.0 255.240.0.0\", \"192.168.100.0 255.255.255.0\"]).  These will be pushed to the clients."
+  description = "A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. [\"10.224.0.0 255.240.0.0\", \"192.168.100.0 255.255.255.0\"]).  This will be concatenated with the list of S3 gateway endpoint routes and these will be pushed to the clients."
```
```suggestion
  description = "A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. [\"10.224.0.0 255.240.0.0\", \"192.168.100.0 255.255.255.0\"]).  This will be concatenated with the list of S3 gateway endpoint routes and the result will be pushed to the clients."
```

jsf9k reviewed (approved):
I have one suggested language change, which you can take or leave.  Approved with gusto!

dav3r commented on locals.tf:
```diff
@@ -29,4 +29,11 @@ locals {
+
+  # Turn the prefix list CIDRS for the S3 gateway endpoint into a list of OpenVPN
+  # friendly "network netmask" entries.
+  private_networks = [
```
Do we want to call this local variable something different to avoid confusion with `var.private_networks`?  Maybe just `vpc_endpoint_cidrs`?

jsf9k replied:
That's a good suggestion.

mcdonnnj replied:
Since we deconstruct the CIDR into a network and netmask is there a more accurate description? My main issue is that since they are not in CIDR notation I do not want the variable name to cause a mistaken assumption.

mcdonnnj replied:
I think this question got lost in the tail end of Friday and I would appreciate input from @cisagov/team-ois.

jsf9k replied:
How about `private_networks_and_s3`?  I agree with @dav3r that it's quite easy to confuse `local.private_networks` with `var.private_networks`, and that's a good name to give the local a more specific name.

mcdonnnj replied:
Oh I definitely agree with changing the variable name. I just had issues with `vpc_endpoint_cidrs` as the replacement.

dav3r replied:
How about just `vpc_endpoints` then?

mcdonnnj replied:
Updated in 25aa490.

dav3r reviewed (approved):
Approved, but please see my one question. 🎾

mcdonnnj committed:
Update variable description

Update the `private_networks` description to mention that the routes from the concatenation are what is pushed to clients..

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Update a variable name to be more descriptive

Update the local variable name containing the OpenVPN routes for the S3 VPC
endpoints so that it is both more descriptive and less likely to be confused
with an existing input variable.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>
